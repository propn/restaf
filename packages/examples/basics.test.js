let testFunctions = require("./testFunctions");
let testInfo;
beforeAll(async () => {
  try {
    testInfo = await require('./lib/setupAll')();
  }
  catch (err) {
    console.log(err);
    process.exit(1);
  }
});
test("logon and get root links for default services", async () => {
  expect.assertions();
  let s = process.env.DEFAULT_VIYA_SERVICES.split(',').map(s1 => {
    return s1.trim();
  });
  expect.assertions();

  let l = await testFunctions.addServices(s, testInfo);
  expect(l).toMatchSnapshot();
});
test('logon and get root links for VIYA_SERVICES env variable', async () => {
  if (process.env.VIYA_SERVICES != null) {
    expect.assertions();
    let s = process.env.VIYA_SERVICES.split(',').map(s1 => {
      return s1.trim();
    });
    let l = await testFunctions.addServices(s, testInfo);
    expect(l).toEqual(s);
  }
});