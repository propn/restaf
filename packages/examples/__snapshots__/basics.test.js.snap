// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logon and get root links for default services 1`] = `
Array [
  "reports",
  "reportImages",
  "reportTransforms",
  "compute",
  "files",
  "casManagement",
  "modelPublish",
  "modelRepository",
  "jobExecution",
  "casProxy",
]
`;
